tests/test_execution.py

```diff
@@ -10,14 +10,11 @@
 
 
 @pytest.mark.slow
-def test_tuples_execution_on_same_node(global_execution_env):
+def test_tuples_execution_on_same_node(factory, client, data_envs_1):
     """Execution of a traintuple, a following testtuple and a following traintuple."""
-    factory, initial_assets, network = global_execution_env
-    client = network.clients[0]
 
-    initial_assets = initial_assets.filter_by(client.node_id)
-    dataset = initial_assets.datasets[0]
-    objective = initial_assets.objectives[0]
+    dataset = data_envs_1.datasets[0]
+    objective = data_envs_1.objectives[0]
```

Comment:
In most of the executions test we are instantiating these variables and using them only once.

Thanks to this PR, accessing the first dataset of the node 1 for instance is short and explicit.

We could avoid creating these variables and directly use `data_env_1.<asset_name>s[i]`. It could make the test shorter without decreasing its readability. What do you think @jmorel ?

Reply:
It could be good for most of them, but I think we'll have to keep some variables for those that are more used 🙂

Reply:
Or we could have some more fixtures, `default_dataset` and `default_objective` that would return just these.

Reply:
Sounds like a good idea as each data_env has currently only a single dataset and a single objective.